forest/benchmarking/superoperator_conversion.py

```diff
@@ -63,6 +64,37 @@ def unvec(vector):
     matrix = vector.reshape(dim, dim).T
     return matrix
 
+def kraus2chi(kraus_ops: list):
+    r"""
+    Convert a set of Kraus operators (representing a channel) to
+    a chi matrix which is also known as a process matrix.
+
+    :param kraus_ops: A list or tuple of N Kraus operators
+    :return: Returns a D^2 x D^2 matrix.
+    """
+    if not isinstance(kraus_ops, (list, tuple)):
+        kraus_ops = [kraus_ops]
+
+    n_kraus_ops = len(kraus_ops)
+    dim_squared = kraus_ops[0].size
+    n_qubits = int(np.log2(np.sqrt(dim_squared)))
+    n_pauli_ops = dim_squared # reminder
+
+    # decompose Kraus operators into Pauli basis and store coeffs c_{k,p}
+    c_k_p = np.zeros((n_kraus_ops, n_pauli_ops), dtype=complex)
+    for kdx, krausop in enumerate(kraus_ops):
+        for pdx, Pauli in enumerate(n_qubit_pauli_basis(n_qubits)):
+            c_k_p[kdx, pdx] = np.trace(np.matmul(Pauli[1], krausop))
+
+    # multiply coeffs to get process matrix elements
+    chi_mat = np.zeros((dim_squared, dim_squared), dtype=complex)
+    for idx, jdx in itertools.product(range(0, n_pauli_ops), range(0, n_pauli_ops)):
+        chi_ij = 0
+        # \chi_{i, j} = \sum_k c_{k, i} c_{k, j} ^ *
+        for kdx in range(0, n_kraus_ops):
+            chi_ij += c_k_p[kdx, idx] * np.conjugate(c_k_p[kdx, jdx])
+        chi_mat[idx, jdx] = chi_ij / dim_squared
+    return chi_mat
```

Comment:
Can't we just do 
c_vecs = [computational2pauli_basis_matrix * vec(kraus) for kraus in kraus_ops]
chi_mat = sum( [np.kron(c_vec, c_vec.conj().T for c_vec in c_vecs]) 

I think that's basically what you are doing? This includes the factor of 1/dim_squared that is hidden in c2p_basis_matrix that I mentioned above.

Reply:
Thanks!
* -> matmul
kron -> matmul